.github/workflows/ci-wordpress.yaml

```diff
@@ -0,0 +1,98 @@
+#
+# Original Author: Rostislav Udaltsov
+# Created: 09/19/2023
+# ----------------------------------------------------------------------------------------------------------
+
+on:
+  workflow_call:
+    inputs:
+      # Build configuration
+      server_host:
+        required: true
+        type: string
+        description: 'The address of the server'
+      environment:
+        required: true
+        type: string
+        description: 'The environment'
+      playbook_path:
+        required: true
+        type: string
+        description: 'The path to the file relative to the repository'
+      deploy_path:
+        required: true
+        type: string
+        description: 'The path to deploy on the production server'
+      deploy_repo:
+        required: true
+        type: string
+        description: 'The repository for deploy'
+      deploy_branch:
+        required: true
+        type: string
+        description: 'The branche for deploy'
+      python_version:
+        required: true
+        type: number
+        description: 'The version of python'
+      runner_image:
+        required: true
+        type: string
+        description: 'The image of the runner OS on which workflow is running'
+    # Secrets for ssh
+    secrets:
+      ssh_deploy_user:
+        required: true
+        description: 'The username for connect to production server via ssh'
+      ssh_private_key:
+        required: true
+        description: 'The private key for connect to production server via ssh'
+
+jobs:
+  deploy:
+    runs-on: ${{ inputs.runner_image }}
+
+    steps:
+      # Preparing workspace: install python, ansible and ansistrano-role
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ inputs.PYTHON_VERSION }}
+
+      - name: Install Ansible
+        run: |
+          python -m pip install --upgrade pip
+          pip install ansible requests jinja2
+
+      - name: Install Ansistrano roles
+        run: |
+          ansible-galaxy install ansistrano.deploy ansistrano.rollback
+
+      # Adding private-key to ssh-agent
+      - name: Set up ssh-agent
+        uses: webfactory/ssh-agent@v0.5.2
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+
+      # Deploy the project
+      - name: Ansistrano deploy
+        env:
+          # Environment variables for ansible must be in uppercase
+          ANSIBLE_HOST_KEY_CHECKING: False
+          ANSIBLE_GATHERING: 'smart'
+          ANSIBLE_PIPELINING: True
+          ANSIBLE_REMOTE_USER: ${{ secrets.SSH_DEPLOY_USER }}
+        run: |
+          ansible-playbook -i \
+          ${{ inputs.SERVER_HOST }} \
+          ${{ inputs.PLAYBOOK_PATH }} \
+          --extra-vars \
+          'deploy_path=${{ inputs.DEPLOY_PATH }} \
+           deploy_repo=${{ inputs.DEPLOY_REPO }} \
+           deploy_branch=${{ inputs.DEPLOY_BRANCH }}'
```

Comment:
Why don't use [default variables](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) like `GITHUB_ACTION_REPOSITORY` and `GITHUB_BASE_REF`?

Reply:
The github context contains a `repositoryUrl` variable in which the repository address is represented as `git://github.com/saritasa-nest/ceai-wordpress.git`. This option is not suitable for ansible, so a variable with an explicit indication of the address is used.
When using the `github.ref_name variable`, the current branch will be substituted, and for example we want to use the contents of branch `develop` or `main`, so I explicitly specified the name of the branch.

Reply:
You can try: `github.server_url`, `github.repository`, `github.ref_name`